From 57741dfbbb6c9bae7050989fdbe242bacb6f92e1 Mon Sep 17 00:00:00 2001
From: Thor Thayer <tthayer@opensource.altera.com>
Date: Tue, 9 Feb 2016 10:21:10 -0600
Subject: [PATCH] FogBugz #355420-2: dts: socfpga: Fix spidev WARN_ON() during
 boot.

A WARN_ON() was added to the spidev framework and the spidef compatible
string was removed. The preference is to use an actual part. In our
case, the SPI port is unused but still needs to be tested with each
release. The fix is to spoof spidev using the existing ROHM DAC
compatible string.

Signed-off-by: Thor Thayer <tthayer@opensource.altera.com>
Signed-off-by: Srikanth Krishnakar <Srikanth_Krishnakar@mentor.com>
---
 arch/arm/boot/dts/socfpga_cyclone5_socdk.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts b/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts
index fea06fa..da4d3f0 100644
--- a/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts
+++ b/arch/arm/boot/dts/socfpga_cyclone5_socdk.dts
@@ -170,7 +170,8 @@
 	status = "okay";
 
 	spidev@0 {
-		compatible = "spidev";
+		/* spidev causes a WARN_ON() so spoof with DAC compat */
+		compatible = "rohm,dh2228fv";
 		reg = <0>;	/* chip select */
 		spi-max-frequency = <1000000>;
 		enable-dma = <1>;
-- 
2.7.4

